### ISSUE TITLE
Request for Processed Data

### ISSUE THREAD

**Issue by MeycL** [Liuwq-bit/LightGT#1]
Thank you so much for your wonderful work, heading over can you provide the processed data?

**Comment by Liuwq-bit** [comment]
Thank you for your attention.

Due to the copyright, I cannot release the data set directly, please go to the official website to apply.

In this paper, I am responsible for writing the implementation code of this model, and the code of the data processing part is written by another collaborator.

With the consent of that collaborator, I put some of the code below. You can process the data yourself by referring to the code after applying for the complete data set.

~~~
import torch
import numpy as np
import scipy.sparse as sparse
from sklearn.decomposition import TruncatedSVD
import torch.nn.functional as F
from scipy.sparse import coo_matrix

ITEM_NUM = 76085
WORD_NUM = 11574

T = torch.load('./tiktok_data/textual_feat_new.pt')
# print(T.size())

T = T.numpy()
print(T)


print(np.unique(T[0]))
print(np.unique(T[1]))

np.save('../../share/user0/tiktok/pro_data/tiktok_t_lookup.npy', T)

row = T[0]
# print(T[[0, 1]])
col = T[1]
data = np.ones_like(row)
item_text = sparse.coo_matrix((data, (row, col)), shape=(ITEM_NUM, WORD_NUM), dtype=np.int8)
# item_text.data = np.array(item_text.data, dtype=np.int8)
item_text = item_text.toarray()  # the 0-1 matrix of testing set

svd = TruncatedSVD(n_components=4)
title_signal_lookup = svd.fit_transform(item_text)

print(title_signal_lookup.shape)

np.save('../../share/user0/tiktok/pro_data/tiktok_t_s.npy', title_signal_lookup)

print(item_text.sum())

print(title_signal_lookup)
# item_text_tensor = torch.from_numpy(item_text)

# item_text_tensor = F.normalize(item_text_tensor.float(), p=1, dim=1).numpy()
# SS = coo_matrix(item_text_tensor)
# TT = np.concatenate(([SS.row], [SS.col], [SS.data]), axis=0)
# print(TT.shape)

# np.save('../../share/user0/tiktok/pro_data/tiktok_t_lookup.npy', T)

print('finish...')
~~~

**Comment by Liuwq-bit** [comment]
Thank you for your attention.

Due to the copyright, I cannot release the data set directly, please go to the official website to apply.

In this paper, I am responsible for writing the implementation code of this model, and the code of the data processing part is written by another collaborator.

With the consent of that collaborator, I put some of the code below. You can process the data yourself by referring to the code after applying for the complete data set.

~~~
import torch
import numpy as np
import scipy.sparse as sparse
from sklearn.decomposition import TruncatedSVD
import torch.nn.functional as F
from scipy.sparse import coo_matrix

ITEM_NUM = 76085
WORD_NUM = 11574

T = torch.load('./tiktok_data/textual_feat_new.pt')
# print(T.size())

T = T.numpy()
print(T)


print(np.unique(T[0]))
print(np.unique(T[1]))

np.save('../../share/user0/tiktok/pro_data/tiktok_t_lookup.npy', T)

row = T[0]
# print(T[[0, 1]])
col = T[1]
data = np.ones_like(row)
item_text = sparse.coo_matrix((data, (row, col)), shape=(ITEM_NUM, WORD_NUM), dtype=np.int8)
# item_text.data = np.array(item_text.data, dtype=np.int8)
item_text = item_text.toarray()  # the 0-1 matrix of testing set

svd = TruncatedSVD(n_components=4)
title_signal_lookup = svd.fit_transform(item_text)

print(title_signal_lookup.shape)

np.save('../../share/user0/tiktok/pro_data/tiktok_t_s.npy', title_signal_lookup)

print(item_text.sum())

print(title_signal_lookup)
# item_text_tensor = torch.from_numpy(item_text)

# item_text_tensor = F.normalize(item_text_tensor.float(), p=1, dim=1).numpy()
# SS = coo_matrix(item_text_tensor)
# TT = np.concatenate(([SS.row], [SS.col], [SS.data]), axis=0)
# print(TT.shape)

# np.save('../../share/user0/tiktok/pro_data/tiktok_t_lookup.npy', T)

print('finish...')
~~~

**Comment by MeycL** [comment]
But the link you gave, doesn't have the tiktok dataset anymore.

**Comment by Liuwq-bit** [comment]
Sorry, our tiktok dataset came from that link. Now I don't know where to find this data set.

**Comment by MeycL** [comment]
So can I please send the dataset used for the experiment to this email address (2641457512@qq.com)? I promise I won't give it away, and more importantly, I need to compare it with this algorithm of yours for my research and I need to make sure how it works?

**Comment by MeycL** [comment]
So can you please send the dataset used for the experiment to this email address ([2641457512@qq.com](mailto:2641457512@qq.com))? I promise I won't give it away, and more importantly, I need to compare it with this algorithm of yours for my research and I need to make sure how it works?

**Comment by Liuwq-bit** [comment]
Sorry, I can't provide it to you according to the copyright agreement. Please apply through the relevant channels.

**Comment by MeycL** [comment]
OK, I get it, I'll just give up.
